### PR TITLE
Add unit tests for version

### PR DESCRIPTION
**Pull request by ppinchuk**
## Pull Request Overview
Introduce unit tests to verify the presence and semantic correctness of the `__version__` attribute in the compass module. These tests ensure compliance with the expected versioning format.

### Notes
Just a few quick tests. Not super important but might help us in the long run to catch tricky edge case issues, especially when running tests on some remote mahcines

### CODE REVIEW

File: tests/python/unit/test_version.py [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+We allow dev/local forms produced by setuptools_scm
+(e.g., 0.11.3.dev8+gHASH).
+"""
+
+import re
+
+import compass
+
+
+SEMVER_DEV_PATTERN = re.compile(r"^\d+\.\d+\.\d+(?:\.dev\d+\+g[0-9a-f]+)?$")
```
Comment (castelao) [inline tests/python/unit/test_version.py]: We haven't been using, but should we allow for release candidates (`rc`) as well? I think we will start using that in the future.

I don't have strong opinion about this. We can just wait

**Review by castelao — COMMENTED** [review]
Is the idea here to capture a bad pipeline that would result in a bad setup, such as a docker container built with the wrong version?

File: tests/python/unit/test_version.py [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+    assert compass.__version__  # non-empty
+
+
+def test_version_semantic_shape():
+    """Version matches dev semver pattern"""
+    v = compass.__version__
+    assert SEMVER_DEV_PATTERN.match(v) is not None, (
+        f"Version {v} does not match expected pattern"
+    )
+    assert v != "9999", "Version set to placeholder"
```
[the remark this answers is not in the document]
Comment (ppinchuk) [reply tests/python/unit/test_version.py]: I'm not sure actually. Doesn't it depend on if you download with git available? Like if i `git clone` the repo but manually delete all the tags or something, then I would expect it to be `10000` (and extra). But if I just downloaded a zip file of the code and put it under some directory with no git attached, would it go to `9999`? I guess I'll have to check this....

[the remark this answers is not in the document]
Comment (ppinchuk) [reply tests/python/unit/test_version.py]: Yup and when I download a zip file and extract it (no git attached), I get this: 

image

**Comment by ppinchuk** [comment]
I guess the downside to this test is that all our actions now have to run with a `fetch-depth: 0` flag. I don't think that's too big of a deal, but I'm also not sure attached to this test if you think this would be a pain @castelao

File: tests/python/unit/test_version.py (hunk 1 above)
Comment (castelao) [reply tests/python/unit/test_version.py]: I would use `rc` for the final checks. Share the `rc` wheels with the team and maybe use it for some short time to confirm before releasing so we would give more resilience for everyone using `compass`. That said, it would be just fine to use only `dev`, so let's keep only `dev` for now.

File: tests/python/unit/test_version.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (castelao) [reply tests/python/unit/test_version.py]: > So I will add a check for both cases

What about `>=9999`? I do hope we never release that many versions of `compass`!

**Comment by castelao** [comment]
> I guess the downside to this test is that all our actions now have to run with a `fetch-depth: 0` flag. I don't think that's too big of a deal, but I'm also not sure attached to this test if you think this would be a pain @castelao

I agree, I don't think it is a big deal to require `fetch-depth: 0`. Currently, the most important place to verify/check the version is when building the container, which we don't verify that today. We should attach the version of the container to the version of the library, so we can still play with the intermediate dev release but also be clear on what are we deploying in production.

I don't see a clear benefit with this test here, but if you want to add it, I support you.

File: tests/python/unit/test_version.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (castelao) [reply tests/python/unit/test_version.py]: Duh! Or course!
